### PR TITLE
Allow users to remove secondary emails

### LINKED ISSUE #365: Ability to remove secondary email addresses
In gitlab by @1219693 on Jan 28, 2016, 21:09

Addition of the ability for users to remove secondary email addresses.

### PR DESCRIPTION
**Pull request by atoff**
As above. Allows users to delete their secondary emails. Set up a cascade to delete these emails sso assignments as well.

Closes #365

### CODE REVIEW

File: app/Http/Controllers/Mship/Management.php [hunk 1]
```diff
@@ -73,6 +73,32 @@ public function postEmailAdd()
                        ->withSuccess('Your new email ('.$email.') has been added successfully! You will be sent a verification link to activate this email address.');
     }
 
+    public function getEmailDelete(AccountEmail $email)
+    {
+        // Is this the user's email?
+        if ($email->account->id != $this->account->id) {
```
Comment (NFarrington) [inline app/Http/Controllers/Mship/Management.php]: Ideally these should use strict comparisons: !==

File: resources/views/mship/management/email/delete.blade.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <div class="panel-heading"> Delete Secondary Email</div>
+    <div class="panel-body text-center">
+      <!-- Content Of Panel [START] -->
+      <p>
+        Are you sure you would like to delete <b>{{ $email->email }}</b> from your secondary emails?
+      </p>
+
+      @if ($assignments->count() > 0)
+        <p>
+          Deleteing this email will reset the following SSO assignments to their default value ({{ \Auth::user()->email }})
```
Comment (NFarrington) [inline resources/views/mship/management/email/delete.blade.php]: > Deleteing

(!sp)

File: app/Http/Controllers/Mship/Management.php [hunk 3]
```diff
@@ -73,6 +73,32 @@ public function postEmailAdd()
...
+
+        return $this->viewMake('mship.management.email.delete')
+                    ->with('email', $email)
+                    ->with('assignments', $email->ssoEmails);
+    }
+
+    public function postEmailDelete(AccountEmail $email)
+    {
+        // Is this the user's email?
+        if ($email->account->id != $this->account->id) {
```
Comment (NFarrington) [inline app/Http/Controllers/Mship/Management.php]: Ideally these should use strict comparisons: !==

**Comment by atoff** [comment]
Thanks Neil, these corrections have been made

**Pull request merged**